### PR TITLE
Asynchronous loading of tables by default

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Backward Incompatible Change

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Enable asynchronous load of databases and tables by default. See the `async_load_databases` in config.xml. While this change is fully compatible, it can introduce a difference in behavior. When `async_load_databases` is false, as in the previous versions, the server will not accept connections until all tables are loaded. When `async_load_databases` is true, as in the new version, the server can accept connections before all the tables are loaded. If a query is made to a table that is not yet loaded, it will wait for the table's loading, which can take considerable time. It can change the behavior of the server if it is part of a large distributed system under a load balancer. In the first case, the load balancer can get a connection refusal and quickly failover to another server. In the second case, the load balancer can connect to a server that is still loading the tables, and the query will have a higher latency. Moreover, if many queries accumulate in the waiting state, it can lead to a "thundering herd" problem when they start processing simultaneously. This can make a difference only for highly loaded distributed backends. You can set the value of `async_load_databases` to false to avoid this problem.

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit ee1b3042848b0dc9c1362e4dce89a44bb7f8abbe with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57695/ee1b3042848b0dc9c1362e4dce89a44bb7f8abbe/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/7145687584">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57695/ee1b3042848b0dc9c1362e4dce89a44bb7f8abbe/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: programs/server/config.xml [hunk 1]
```diff
@@ -371,7 +371,7 @@
     <!-- Enables asynchronous loading of databases and tables to speedup server startup.
          Queries to not yet loaded entity will be blocked until load is finished.
       -->
-    <!-- <async_load_databases>true</async_load_databases> -->
+    <async_load_databases>true</async_load_databases>
```
Comment (nikitamikhaylov) [inline programs/server/config.xml:374]: Let move it to `tests/config/config.d`.

Comment (alexey-milovidov) [reply programs/server/config.xml:374]: My intent is to enable it by default.

**Comment by alesapin** [comment]
@alexey-milovidov Isn't this change backward incompatible? Now sever will start even with unavailable tables. Proxies and balancers will route queries to this server, queries will start to fail with exceptions. I suspect that for many users it will breaking behavior.

**Comment by alexey-milovidov** [comment]
@alesapin, we can highlight it as a backward incompatible change.

**Comment by alexey-milovidov** [comment]
@alesapin Added description.

**Pull request merged**